nbformat/sign.py

```diff
@@ -123,8 +123,17 @@ def _db_default(self):
             self.log.warn("Missing SQLite3, all notebooks will be untrusted!")
             return
         kwargs = dict(detect_types=sqlite3.PARSE_DECLTYPES|sqlite3.PARSE_COLNAMES)
-        db = sqlite3.connect(self.db_file, **kwargs)
-        self.init_db(db)
+        try:
+            db = sqlite3.connect(self.db_file, **kwargs)
+            self.init_db(db)
+        except sqlite3.DatabaseError:
+            self.log.warn("Issues opening database: Moving the .db file to .db.bak and creating a new one.  Use jupyter --data-dir to find the old file.")
```

Comment:
Don't forget that the code knows where the database is - you can construct the string with the actual location in!

The message should also tell the user:
* That it means the notebook signatures database. 'database' isn't very specific!
* What this means, besides moving a file. This database is how we track 'trusted' notebooks that can run Javascript as soon as you open them. If it's got corrupted, that means all notebooks will be untrusted again, so you might have to re-run notebooks in order to see the output (or explicitly mark them as trusted). Let's see how clear and concise we can make that. ;-)